src/Foundation/Http/Middleware/CheckForTrustedHost.php

```diff
@@ -0,0 +1,79 @@
+<?php namespace October\Rain\Foundation\Http\Middleware;
+
+use Config;
+use Illuminate\Http\Middleware\TrustHosts as BaseMiddleware;
+
+class CheckForTrustedHost extends BaseMiddleware
+{
+    /**
+     * Get the host patterns that should be trusted.
+     *
+     * Trusted hosts should be defined in the `config/app.php` configuration file as an array, ie.:
+     *
+     *   'trustedHosts' => [
+     *      'example.com',           // Matches just example.com
+     *      'www.example.com',       // Matches just www.example.com
+     *      '^(.+\.)?example\.com$', // Matches example.com and all subdomains
+     *      'https://example.com',   // Matches just example.com
+     *   ],
+     *
+     * or as a boolean - if true, it will trust the `app.url` host and all subdomains, if false it
+     * will disable the feature entirely.
+     *
+     * Hosts can be defined as regex patterns for complex matching.
+     *
+     * @return array
+     */
+    public function hosts()
+    {
+        return self::processTrustedHosts(Config::get('app.trustedHosts', []));
+    }
+
+    /**
+     * Processes the trusted hosts into an array of patterns the match for host header checks.
+     *
+     * @param array|bool $hosts
+     * @return array
+     */
+    public static function processTrustedHosts($hosts)
+    {
+        if ($hosts === true) {
+            $url = Config::get('app.url', null);
+
+            // If no app URL is set, then disable trusted hosts.
+            if (is_null($url)) {
+                return [];
+            }
+
+            $hosts = [$url];
+        } elseif ($hosts === false) {
+            return [];
+        }
+
+        $hosts = array_map(function ($host) {
+            // If a URL is provided, extract the host
+            if (filter_var($host, FILTER_VALIDATE_URL)) {
+                $host = parse_url($host, PHP_URL_HOST);
+            }
+
+            // Do strict checks for IP address hosts
+            if (filter_var($host, FILTER_VALIDATE_IP)) {
+                return '^' . preg_quote($host) . '$';
+            }
+
+            // Allow domains to be matched against a www subdomain
+            if (filter_var($host, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME)) {
+                if (preg_match('/^www\.(.*?)$/i', $host, $matches)) {
```

Comment:
Could you explain what this regex is doing here? Also, I don't think that we should be automatically adding the www subdomain in any case except for trustedHosts = true, if people are explicitly defining hosts with an array then they shouldn't be having those explicit definitions automatically changed at all.

Reply:
Fair point - will change it.

Reply:
The regex here was checking if the given domain starts with `www`, and if so, to allow the domain without the `www` subdomain, given that quite a few hosts (most notably cPanel) will automatically set both to point to the same site.